source/common/config/datasource.cc

```diff
@@ -64,24 +66,12 @@ RemoteAsyncDataProvider::RemoteAsyncDataProvider(
       retries_remaining_(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(source.retry_policy(), num_retries, RetryCount)) {
 
-  uint64_t base_interval_ms = RetryInitialDelayMilliseconds;
-  uint64_t max_interval_ms = RetryMaxDelayMilliseconds;
-  if (source.has_retry_policy()) {
-    if (source.retry_policy().has_retry_back_off()) {
-      base_interval_ms =
-          PROTOBUF_GET_MS_REQUIRED(source.retry_policy().retry_back_off(), base_interval);
-
-      max_interval_ms = PROTOBUF_GET_MS_OR_DEFAULT(source.retry_policy().retry_back_off(),
-                                                   max_interval, base_interval_ms * 10);
-
-      if (max_interval_ms < base_interval_ms) {
-        throw EnvoyException("max_interval must be greater than or equal to the base_interval");
-      }
-    }
+  backoff_strategy_ = Utility::prepareJitteredExponentialBackOffStrategy(source, random);
```

Comment:
You can pass `RetryInitialDelayMilliseconds` and `RetryMaxDelayMilliseconds` as optionals to the `prepareJitteredExponentialBackOffStrategy` function, so it will always return a valid strategy, and will allow one to override the defaults.

Reply:
done, thanks!